modules/basis_universal/register_types.cpp

```diff
@@ -52,44 +52,50 @@ enum BasisDecompressFormat {
 #ifdef TOOLS_ENABLED
 static Vector<uint8_t> basis_universal_packer(const Ref<Image> &p_image, Image::UsedChannels p_channels) {
 	Vector<uint8_t> budata;
-
 	{
+		basisu::basis_compressor_params params;
 		Ref<Image> image = p_image->duplicate();
-
-		// unfortunately, basis universal does not support compressing supplied mipmaps,
-		// so for the time being, only compressing individual images will have to do.
-
-		if (image->has_mipmaps()) {
-			image->clear_mipmaps();
-		}
 		if (image->get_format() != Image::FORMAT_RGBA8) {
 			image->convert(Image::FORMAT_RGBA8);
 		}
-
-		basisu::image buimg(image->get_width(), image->get_height());
-
+		Ref<Image> image_single = image->duplicate();
 		{
-			Vector<uint8_t> vec = image->get_data();
+			if (image_single->has_mipmaps()) {
+				image_single->clear_mipmaps();
+			}
+			basisu::image buimg(image_single->get_width(), image_single->get_height());
+			Vector<uint8_t> vec = image_single->get_data();
 			const uint8_t *r = vec.ptr();
-
 			memcpy(buimg.get_ptr(), r, vec.size());
+			params.m_source_images.push_back(buimg);
+		}
+		basisu::vector<basisu::image> source_images;
+		for (int32_t mipmap_i = 1; mipmap_i < image->get_mipmap_count(); mipmap_i++) {
+			Ref<Image> mip = image->get_image_from_mipmap(mipmap_i);
+			basisu::image buimg(mip->get_width(), mip->get_height());
```

Comment:
Please test with non-power of two and non-multiple of 4 to see if anything needs to be changed.

While LOD 0 must be multiple of 4, it is legal for mipmaps to be non-multiple of 4. Make sure basis supports non-padded non-multiple of 4 mips properly. it may need to be padded depending on the API

Reply:
The current code forces power of two for all basisu textures.